kore/app/parser/Main.hs

```diff
@@ -99,21 +99,22 @@ main = handleTop $ do
 
             let KoreParserOptions { patternOpt } = koreParserOptions
             for_ patternOpt $ \patternOptions -> do
-                let PatternOptions { patternFileName } = patternOptions
-                parsedPattern <- mainPatternParse patternFileName
-                when willVerify $ do
-                    let PatternOptions { mainModuleName } = patternOptions
-                    indexedModule <-
-                        lookupMainModule
-                            (ModuleName mainModuleName)
-                            indexedModules
-                        & lift
-                    _ <- mainPatternVerify indexedModule parsedPattern
-                    return ()
-                let KoreParserOptions { willPrintPattern } =
-                        koreParserOptions
-                when willPrintPattern $
-                    lift $ putDoc (debug parsedPattern)
+                let PatternOptions { mainModuleName } = patternOptions
+                let PatternOptions { patternFileNames } = patternOptions
+                for_ patternFileNames $ \patternFileName -> do
+                    parsedPattern <- mainPatternParse patternFileName
+                    when willVerify $ do
+                        indexedModule <-
+                            lookupMainModule
+                                (ModuleName mainModuleName)
+                                indexedModules
+                            & lift
```

Comment:
We only need to look up the main module once, so we should move this out of the loop.

After that, let's extract the remaining body of the loop as a top-level function `verifyPatternFile`.

Reply:
I actually had this snippet outside the loop originally but I had to change it after I realised that an error would be thrown if we ran that code with `willVerify` set to `False` (because `indexedModules` would be empty).
I'll push a workaround, though it won't be very pretty due to the control flow that we're going for...

Reply:
Also without that `indexedModule...` snippet, the body of the loop is quite small and straightforward...
I feel like if we isolated that as a separate top-level function we would have to pass on to it a lot of local variables that at the moment are implicitly in scope... (`indexedModule`, `willVerify`, `patternFileName`, `koreParserOptions`)
So I am leaning more towards not isolating it, what do you think?

Reply:
I cleaned up the application logic. Feel free to merge it when you are satisfied.